modules/seqexec/server/src/main/scala/seqexec/server/gmos/Gmos.scala

```diff
@@ -63,17 +64,20 @@ abstract class Gmos[F[_]: Sync, T<:GmosController.SiteDependentTypes](controller
   // The OT lets beams up to G but in practice it is always A/B
   private val BeamLabels = List('A, 'B, 'C, 'D, 'E, 'F, 'G)
 
+  def configToAngle(s: String): Either[ExtractFailure, Angle] =
+    s.parseDoubleOption.toRight(ContentError("Invalid offset value")).map(Angle.fromDoubleArcseconds(_))
+
   private def nsPosition(config: Config, sc: Int): Either[ExtractFailure, Vector[NSPosition]] = {
     (for {
       i <- 0 to scala.math.min(BeamLabels.length, sc) - 1
     } yield {
       for {
-        s <- BeamLabels.lift(i).toRight(ContentError(s"Unknwon label at position $i"))
-        p <- config.extractAs[String](INSTRUMENT_KEY / s"nsBeam${s.name}-p").map(_.toDouble)
-        q <- config.extractAs[String](INSTRUMENT_KEY / s"nsBeam${s.name}-q").map(_.toDouble)
+        s <- BeamLabels.lift(i).toRight(ContentError(s"Unknown label at position $i"))
+        p <- config.extractAs[String](INSTRUMENT_KEY / s"nsBeam${s.name}-p").flatMap(configToAngle).flatMap(Offset.P(_).asRight)
+        q <- config.extractAs[String](INSTRUMENT_KEY / s"nsBeam${s.name}-q").flatMap(configToAngle).flatMap(Offset.Q(_).asRight)
```

Comment:
Once you have an `Angle` the `Offset.Q` constructor can't fail so it seems like `.flatMap(configToAngle).map(Offset.Q(_))` would be sufficient?